hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -173,6 +173,10 @@ public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Opti
 
   public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Option<Map<String, String>> extraMetadata,
                              String commitActionType, Map<String, List<String>> partitionToReplaceFileIds) {
+    // Skip the empty commit
+    if (stats.isEmpty()) {
```

Comment:
Will this check impact the performance of non-empty commit? Will the previous stages being cached?
I personally feel it's better to do the empty check before run into hudi's operation.

Reply:
Hi @garyli1019 , it is just a java list#isEmpty here ,so it should has no impact for the performance. In our previous implement, we use the RDD#isEmpty to skip the empty commit, It is a heavy operation.

Reply:
What I was not sure about is if this will change the Spark DAGs. But looks like it will not. 
@vinothchandar @nsivabalan can you guys take a look as well?

Reply:
Guess this is code path when auto commit is disabled. [Here](https://github.com/apache/hudi/blob/647e9faf255b9ae5796f513d13c769a8c87788d4/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java#L237) is the path when autoCommit is set to true. May be we need to fix here as well. But would be nice if we clean the commit property rather than just returning( .requested, .inflight etc). Basically rollback this pending commit. 
but let's hear @vinothchandar thoughts as well whether can we delay an empty dataset to this extent or should we fail fast before itself.

Reply:
I think there was an explicit ask to allow the empty commit before. Lets take deltastreamer which stores the offset of the kafka checkpoints in the commit metadata. If we don't commit when stats are empty the checkpoint will never advance. The transformer  in delta streamer could filter out all records read in that batch for e.g and lead to an empty commit. but the kafka offsets would have advanced. So its not good to do this IMO

Reply:
lets control this using a new config?

Reply:
yes, it is better for this.  And by default we can allow empty commits.